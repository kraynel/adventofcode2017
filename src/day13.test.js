const { getSeverity, getDelay } = require("./day13");
const fs = require("fs");

describe("getSeverity", () => {
  it("should return the severity", () => {
    expect(
      getSeverity(`0: 3
1: 2
4: 4
6: 4`)
    ).toBe(24);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day13.txt", "utf8", (err, data) => {
      expect(getSeverity(data)).toMatchSnapshot();
      done();
    });
  });
});

describe("getDelay", () => {
  it("should return the delay", () => {
    expect(
      getDelay(`0: 3
1: 2
4: 4
6: 4`)
    ).toBe(10);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day13.txt", "utf8", (err, data) => {
      expect(getDelay(data)).toMatchSnapshot();
      done();
    });
  });
});
