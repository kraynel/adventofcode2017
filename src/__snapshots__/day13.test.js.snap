// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDelay should get the value 1`] = `3946838`;

exports[`getSeverity should get the value 1`] = `1728`;
